Readme (#155)
Clinical-Genomics/clinical_EPPs: README.md

# clinical_EPPs 
## About

The External Program Plug-in (EPP) is a script that is configuerd to be run from within a lims step.

Clinical Genomics LIMS is using both scripts that are developed and maintained by Genologics, and scripts that are developed by developers at Clinical Genomics. Scripts developed and maintained by Clinical Genomics are collected in this repocitory.

## Branching model

clinical_EPPs is using github flow branching model as described in our development manual.

## Production and Stage

> Development of new EPPs is preferably be done locally but the final testing is done on the stage server.

Reviewer: ```suggestion
Development of new EPPs is preferably done locally but the final testing is done on the stage server.
```

The production lims system is set up on hippocampus and the stage lims system is set up on amygdala.

ssh into the servers:

`ssh gls@clinical-lims-stage.scilifelab.se`

`ssh gls@clinical-lims.scilifelab.se`

You will need a password wich is kept in the safety locker at clinical genomics.

## Insalling
The procedure for installing is the same on both servers.

clinical_EPPs is cloned into `/home/glsai/opt/` and installed by the glsai user under the conda environment epp_master.

```
sudo -iu glsai
source activate epp_master

cd /home/glsai/opt/clinical_EPPs
git pull <branch name>
python setup.py install

```
the branch that has been installed is now avalibe from within the [lims web interface](https://clinical-lims-stage.scilifelab.se/clarity/).

## Setting up a new EPP

> The branch with the new script has been installed and you want ot test the script through the web interface. (Or deploy it on production. The procedure is the same.)

Reviewer: ```suggestion
The branch with the new script has been installed and you want to test the script through the web interface. (Or deploy it to production. The procedure is the same.)
```

> Say the new script is bcl2fastq.py. Running it from the commandline looks like this:

Reviewer: ```suggestion
Let us call the new script we want to test: `bcl2fastq.py`. Running it from the command line looks like this:
```

```
(epp_master)glsai@clinical-lims-stage:~/opt/clinical_EPPs/EPPs$ python bcl2fastq.py --help

usage: bcl2fastq.py [-h] [-p PID] [-l LOG]

optional arguments:
  -h, --help  show this help message and exit
  -p PID      Lims id for current Process
  -l LOG      File name for standard log file, for runtime information and
              problems.
```

> To make it avalible in the [web interface](https://clinical-lims-stage.scilifelab.se/clarity), go to the CONFIGURATON tab and then select  AUTOMATION. Klick the NEW AUTOMATON button.

Reviewer: ```suggestion
To make it avalible in the [web interface](https://clinical-lims-stage.scilifelab.se/clarity), go to the `CONFIGURATON` tab and then select `AUTOMATION`. Klick the `NEW AUTOMATON` button.
```

- Choose a Automation Name
- Channel Name should always be `limsserver`.
- Enter the command line string. If you need help selecting a token for an argument, klick the TOKENS tab. In this case the string is

> `bash -c "/home/glsai/miniconda2/envs/epp_master/bin/bcl2fastq.py -p {processLuid} -l {compoundOutputFileLuid0}"`

Reviewer: Are {processLuid} just place holder or actual parameters that exits in the env on the servers?
Same with this one: {compoundOutputFileLuid0}

Author: They are parameters that exist. Refered to as TOKENS. They can eg point to the process id to a file placeholder of the process

Reviewer: Could you add a line or two on that because to me it was not obvious that would exist and how they are used

Author: Absolutely. I added a description
